src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -785,6 +785,8 @@ private void HandleMissingBlocksGap(ChainedHeader newTip)
             lock (this.peerLock)
             {
                 this.chainedHeaderTree.FullValidationSucceeded(blockToConnect.ChainedHeader);
+
+                this.chainState.IsAtBestChainTip = this.chainedHeaderTree.IsAtBestChainTip();
```

Comment:
Looks good, and only missing tests

Reply:
Indeed. 
I've added TODO to my list and will add tests after the activation